### PR TITLE
Issue 747 gce provider

### PR DESCRIPTION
**Pull request by petermatulis**

### CODE REVIEW

**Comment by petermatulis** [comment]
Staging:
http://peter.juju.papamike.ca/en/config-gce.html

File: src/en/config-gce.md [hunk 1]
```diff
@@ -1,255 +1,201 @@
...
-    
-      # The GCE provider uses pre-built images when provisioning instances.
-      # You can customize the location in which to find them with the
-      # image-endpoint setting. The default value is the a location within
-      # GCE, so it will give you the best speed when bootstrapping or adding
-      # machines. For more information on the image cache see
-      # https://cloud-images.ubuntu.com/.
-      # image-endpoint: https://www.googleapis.com
-```
+ - The Juju devel PPA (may change) is needed.
```
Comment (evilnick) [inline src/en/config-gce.md]: I would remove this. We obviously can't have this in the 1.25 instructions

File: src/en/config-gce.md [hunk 2]
```diff
@@ -1,255 +1,201 @@
...
 
-!!! Warning: See [GH #747](https://github.com/juju/docs/issues/747) regarding
-configuration issues. This documentation should be fixed soon.
+ - The Juju client (the host running the below commands) will need the ability
+   to contact the GCE infrastructure on TCP ports 22 and 17070.
```
Comment (evilnick) [inline src/en/config-gce.md]: Is this unusual? Juju always needs port 22 AFAIA

File: src/en/config-gce.md [hunk 3]
```diff
@@ -1,255 +1,201 @@
...
 
-To determine the values to add to the configuration, you will
-first have to set up a compatible project in GCE, by following these steps:
+Proceed to install the software.
 
+```bash
+sudo apt-add-repository -y ppa:juju/devel
+sudo apt-get update
+sudo apt-get install -y juju-core
+```
```
Comment (evilnick) [inline src/en/config-gce.md]: so this install is also not needed

File: src/en/config-gce.md [hunk 4]
```diff
@@ -1,255 +1,201 @@
...
 
-## 1. Create a Google Compute Engine project
```
Comment (evilnick) [inline src/en/config-gce.md]: why have you eschewed numbers/steps. people like numbers/steps, and sometimes it makes things easier for us too!

File: src/en/config-gce.md [hunk 5]
```diff
@@ -1,255 +1,201 @@
...
 
-In either case above, you will be prompted to enter some details for the
-project, specifically a name and a project ID. They will both be pre-populated
-with random names, but as it will be useful to distinguish between multiple 
-possible Juju projects, and any other projects you may create, it is a good idea
-to create recognisable names. The `Project ID` needs to be unique, so adding
-some random numbers or a UUID to the end is useful (though you will need to copy
-this elsewhere, so don't make it too unwieldy)
+The *project id* (used later) will be generated automatically. Use the 'Edit'
+button to change it.
```
Comment (evilnick) [inline src/en/config-gce.md]: '' ...to change it" to something more descriptive, e.g. ' Juju-GCE'.

File: src/en/config-gce.md [hunk 6]
```diff
@@ -1,255 +1,201 @@
...
 
-![image showing project details](media/config-gce03.png)
 
+## Google Compute Engine API
 
-## 3. Enable the Google Compute Engine API
+The Google Compute Engine API needs to be enabled for your new project in order
+for Juju to communicate with it. This is done automatically if a "billing
+method" has been set up. By following the below steps you will discover whether
+you need to set up billing or not. Check to see if a trial account is available.
```
Comment (evilnick) [inline src/en/config-gce.md]: do we care about trial accounts? that's up to the user

File: src/en/config-gce.md [hunk 7]
```diff
@@ -1,255 +1,201 @@
...
+shows 'Enable API', clicking it may prompt you to set up a billing method (if
+not already done).
 
-![image showing main screen](media/config-gce04.png)
 
-On the credentials page itself, select 'Create new Client ID'
+## GCE project credentials file
 
-![image showing API page](media/config-gce05.png)
+A collection of credentials-related material will be required. This is obtained
```
Comment (evilnick) [inline src/en/config-gce.md]: I would explain this - i.e., why is it required?
"For Juju to access the Azure API, it needs various credentials. The easiest way of obtaining these... blah blah

File: src/en/config-gce.md [hunk 8]
```diff
@@ -1,255 +1,201 @@
...
+by downloading a file from the UI.
 
-A pop-up will appear asking which type of credential to generate. You *must* 
-select the `Service account` option - Juju needs these credentials to be able to
-create new instances.
+Return to the 'API Manager' and choose the 'Credentials' screen. By default you
+will be within the 'Credentials' tab. Click the 'New credentials' button and
+choose 'Service account key' among the 3 options available:
```
Comment (evilnick) [inline src/en/config-gce.md]: is this quirky Canadian grammar? In blighty, we always choose from, or maybe at a push, chose from among

File: src/en/config-gce.md [hunk 9]
```diff
@@ -1,255 +1,201 @@
...
 
-## 6. Enter the credentials into your environments.yaml file
+`/home/ubuntu/.juju/Juju-GCE-f33a6cdbd8e3.json`
 
-There are two ways to add this information to the environments.yaml file used by
-Juju, both of which are explained here:
+!!! Warning: Due to
+[LP #1533790](https://bugs.launchpad.net/juju-core/+bug/1533790) make a copy of
+the original file and put it in another location. Edit the original file by
+removing the lines containing these keywords: 'project-id', 'auth-uri',
```
Comment (evilnick) [inline src/en/config-gce.md]: are they lines? I thought it was just a blob. Suggest  s/lines/sections for clarity

File: src/en/config-gce.md [hunk 10]
```diff
@@ -1,255 +1,201 @@
...
-Remember that 
-[you can use the `|` character](http://www.yaml.org/spec/1.2/spec.html#id2760844)
-as the first line of a YAML value to interpret all subsequent indented text as a
-continuous string. You will need to strip out line endings and translate any 
-Unicode characters in the string. A simple way to pre-format some of this
-information is to run something like:
+If it does exist (but it was created with an older version of Juju), first move
+it out of the way (back it up) and *then* generate a new one. Alternatively,
+you can output a generic file to screen (STDOUT) and paste the Azure parts into
+your existing file:
```
Comment (evilnick) [inline src/en/config-gce.md]: paste the *GCE* parts...

File: src/en/config-gce.md [hunk 11]
```diff
@@ -1,255 +1,201 @@
...
 
 ```bash
-python -c 'import sys, yaml, json; yaml.safe_dump(json.load(sys.stdin), sys.stdout, default_flow_style=False)' < gce-auth-file.json 
+juju generate-config --show
 ```
 
-In any case, the auth section of the config file should look something like
-this:
+The file will contain a section for the Azure provider.
```
Comment (evilnick) [inline src/en/config-gce.md]: GCE

File: src/en/config-gce.md [hunk 12]
```diff
@@ -1,255 +1,201 @@
...
+### `auth-file`
 
-The credentials do not include the ID of the project you created for Juju to
-use, and this needs to be added to the `environments.yaml` configuration too, 
-under the section commented with `"# Google instance info"`. The project ID is 
-whatever you entered when you created the project. If, by this stage, you have 
-forgotten that ID, you can retrieve it from the 
-[main project page on GCE - https://console.developers.google.com/project](https://console.developers.google.com/project)
+The value of `auth-file` is the path to the credentials file downloaded
+previously.
```
Comment (evilnick) [inline src/en/config-gce.md]: If you numbered things, you could mention the step number...

File: src/en/config-gce.md [hunk 13]
```diff
@@ -1,255 +1,201 @@
...
 
-![image showing project page](media/config-gce07.png)
+### `project-id`
 
+The value of `project-id` is based on the name of the project you created earlier.
```
Comment (evilnick) [inline src/en/config-gce.md]: If you numbered things, you could mention the step number...

File: src/en/config-gce.md [hunk 14]
```diff
@@ -1,255 +1,201 @@
...
+
 
+# Additional notes
 
-## 8. Optional configuration
+See [General configuration options](https://jujucharms.com/docs/stable/config-general)
+for additional and advanced customization of your environment.
+
+
+## gcloud compute CLI tool
```
Comment (evilnick) [inline src/en/config-gce.md]: Just to avoid ugliness, I would suggest " Using the gcloud...

**Comment by evilnick** [comment]
I think that about does it

File: src/en/config-gce.md (hunk 5 above)
Comment (petermatulis) [inline src/en/config-gce.md]: It will automatically be based on the project name.

**Pull request merged**